### PR TITLE
Architecture Documentation

### PR DESCRIPTION
**Pull request by Stijn-FE**
Hi all, I'm Stijn van Es from Secura and as mentioned in issue #208, I wanted to work on the architectural documentation for secureCodeBox (SCB). So far, I have worked mainly on a functional description of SCB and I will continue with the following:

- [ ] A building block overview
- [ ] A deployment overview 
- [ ] A runtime overview
- [ ] A description of relevant cross-cutting concepts

To complete the documentation, some questions regarding ownership and sponsoring have been posed by my colleague @EndPositive in issue secureCodeBox/secureCodeBox#731 on the main repo. Could you please provide some information regarding these questions?

Since I am new to this project I lack some background in the architectural choices and technical debt on this project. Therefore, I won't be able to fill in the chapters on technical debt and quality requirements. Could you please check these chapters on accuracy and maybe fill in the required information.

- [ ] Risks and technical debts
- [ ] Quality Requirements

closes: #208

### CODE REVIEW

**Review by J12934 — COMMENTED** [review]
Hi awesome work.
Really like the structure of the templates, this should really expand the doc to include aspects previously grossly uncovered by it.

I've tried to answered @EndPositive questions in the issue as much as possible, hope this helps.

I've added a suggestion to update the iteratec team members and their roles.

What tool did you use to generate the diagrams? Would it be possible to include their source files in the repo so that they stay editable? Otherwise we'd have to completely redraw them whenever we want to change them.

**Comment by Stijn-FE** [comment]
Hi, happy to hear the structure is to your liking. 

Thanks for the info and the additional changes. 

For the diagrams, I've used [draw.io](https://app.diagrams.net/). The source code is incorporated in the `.png` files. You should be able to load the `.png` files directly into the draw.io editor. I'll add an explanation on how to edit in the docs.

**Comment by Weltraumschaf** [comment]
Hi @Stijn-FE ,

we have decided to start documenting the architecture. The idea is as follows:

1. I review what you've done yet.
2. We merge your PR.
3. I refactor your work to match the [Arc42](https://arc42.org/) template. (This is a widely adopted template for architecture documentation.)
4. Then we evolve the documentation step-by-step.

So, please convert your PR from `draft` to `ready` :-)

If you're not familiar with Arc42 and want to know more about it, we may schedule a short (~30 min) appointment to tell you the basic idea of it.

**Comment by Stijn-FE** [comment]
Hi Sven, 

Good idea to use Arc42. I looked into this a while ago and actually meant to use this template in this PR. If you have any questions regarding the changes in this PR, please let me know.

**Comment by Weltraumschaf** [comment]
Ah, I wondered if you did the structure by accident like Arc42 :-)

File: docs/architecture/introduction.md [hunk 1]
```diff
@@ -1,10 +1,102 @@
...
+# SPDX-FileCopyrightText: 2021 Secura B.V.
 #
 # SPDX-License-Identifier: Apache-2.0
 
 title: "Architecture Introduction"
 sidebar_label: "Introduction"
 ---
 
-> 🔧 Documentation Coming Soon
+This document describes the secureCodeBox Project (SCB) and is based on the [arc42](https://arc42.org/overview/) architecture documentation template. SecureCodeBox is a Kubernetes based, modularized toolchain for continuous security scans of your software project. Its goal is to orchestrate and easily automate a bunch of security-testing tools out of the box. With secureCodeBox we provide a toolchain for continuous scanning of applications to find the low-hanging fruit issues early in the development process and free the resources of the penetration tester to concentrate on the major security issues.
```
Comment (Weltraumschaf) [inline docs/architecture/introduction.md:10]: Ah, you didn't match the Arc42 structure by accident 👍

**Review by Weltraumschaf — CHANGES_REQUESTED** [review]
Nice work! This is a really good starting point and my hope is that we now evolve to a good architecture doc with this foundation :-)

If you do my suggestions (mostly the copyright header) I'll merge this.

**Comment by Weltraumschaf** [comment]
Hi @Stijn-FE,

would you please look into our suggested changes? Our current sprint ends tomorrow and we would like to merge this PR to plan the next steps for the next sprint.

**Comment by EndPositive** [comment]
Leaving a comment here that Stijn is currently unavailable, but verbally agreed to the last few comments.

File: docs/architecture/technical/building-block-view.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+# SPDX-FileCopyrightText: 2021 Secura B.V.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+title: "Architecture Building Block View"
+sidebar_label: "Building Block View"
+---
+
+Below, an overview is given of the global design/architecture of secureCodeBox. This document is inspired by the [C4](https://c4model.com/) model for visualizing software architecture. Conform this model, this document is split in 4 parts (i.e. levels). First, a `context` level overview, displaying the context in which the application is used. The second level, `containers`, broadly describes the different data streams. The third level consists of a `components` overview, describing the different (Django) apps and the interactions between them. The fourth and final level contains a `code` overview. Which will consist of class- and database-diagrams.
```
[the remark this answers is not in the document]
Comment (Weltraumschaf) [reply docs/architecture/technical/building-block-view.md:10]: @EndPositive Maybe you can answer this: Is this a copy & paste error or why Django?

**Comment by Weltraumschaf** [comment]
Hi @Stijn-FE,

@EndPositive told me that you're away at the moment, so I committed all the suggestions and will merge it.

File: docs/architecture/technical/building-block-view.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (EndPositive) [reply docs/architecture/technical/building-block-view.md:10]: Hi Sven, this is indeed a copy paste error. You may remove the text in between brackets and the text looks valid to me.

[the remark this answers is not in the document]
Comment (Weltraumschaf) [reply docs/architecture/technical/building-block-view.md:10]:
```suggestion
Below, an overview is given of the global design/architecture of secureCodeBox. This document is inspired by the [C4](https://c4model.com/) model for visualizing software architecture. Conform this model, this document is split in 4 parts (i.e. levels). First, a `context` level overview, displaying the context in which the application is used. The second level, `containers`, broadly describes the different data streams. The third level consists of a `components` overview, describing the different components and the interactions between them. The fourth and final level contains a `code` overview. Which will consist of class- and database-diagrams.
```

[the remark this answers is not in the document]
Comment (Weltraumschaf) [reply docs/architecture/technical/building-block-view.md:10]: Also replaced apps w/ components. It's not a good sentence, but we do not have "apps" and this text will undergo heavy rewriting in the next months anyway, when we evolve the architecture doc.

**Review by Weltraumschaf — APPROVED** [review]

**Pull request merged**